egret/model_library/transmission/branch.py

```diff
@@ -15,73 +15,116 @@
+    line
+    """
+    decl.declare_var('pfl', model=model, index_set=index_set, **kwargs)
+
+
 def declare_var_pf(model, index_set, **kwargs):
     """
-    Create variable or the real part of the power flow in the "from"
+    Create variable for the real part of the power flow in the "from"
     end of the transmission line
     """
     decl.declare_var('pf', model=model, index_set=index_set, **kwargs)
 
 
 def declare_var_qf(model, index_set, **kwargs):
     """
-    Create variable or the imaginary part of the power flow in the "from"
+    Create variable for the imaginary part of the power flow in the "from"
     end of the transmission line
     """
     decl.declare_var('qf', model=model, index_set=index_set, **kwargs)
 
 
 def declare_var_pt(model, index_set, **kwargs):
     """
-    Create variable or the real part of the power flow in the "to"
+    Create variable for the real part of the power flow in the "to"
     end of the transmission line
     """
     decl.declare_var('pt', model=model, index_set=index_set, **kwargs)
 
 
 def declare_var_qt(model, index_set, **kwargs):
     """
-    Create variable or the imaginary part of the power flow in the "to"
+    Create variable for the imaginary part of the power flow in the "to"
     end of the transmission line
     """
     decl.declare_var('qt', model=model, index_set=index_set, **kwargs)
 
 
 def declare_var_ifr(model, index_set, **kwargs):
     """
-    Create variable or the real part of the current flow in the "from"
+    Create variable for the real part of the current flow in the "from"
     end of the transmission line
     """
     decl.declare_var('ifr', model=model, index_set=index_set, **kwargs)
 
 
 def declare_var_ifj(model, index_set, **kwargs):
     """
-    Create variable or the imaginary part of the current flow in the "from"
+    Create variable for the imaginary part of the current flow in the "from"
     end of the transmission line
     """
     decl.declare_var('ifj', model=model, index_set=index_set, **kwargs)
 
 
 def declare_var_itr(model, index_set, **kwargs):
     """
-    Create variable or the real part of the current flow in the "to"
+    Create variable for the real part of the current flow in the "to"
     end of the transmission line
     """
     decl.declare_var('itr', model=model, index_set=index_set, **kwargs)
 
 
 def declare_var_itj(model, index_set, **kwargs):
     """
-    Create variable or the imaginary part of the current flow in the "to"
+    Create variable for the imaginary part of the current flow in the "to"
     end of the transmission line
     """
     decl.declare_var('itj', model=model, index_set=index_set, **kwargs)
 
 
+def declare_eq_branch_dva(model, index_set, branches):
+    """
+    Create the equality constraints for the angle difference
+    in the branch
+    """
+    m = model
+
+    con_set = decl.declare_set("_con_eq_branch_dva_set", model, index_set)
+
+    m.eq_dva_branch = pe.Constraint(con_set)
```

Comment:
Due to inactive lines some of the indices of the eq_dva_branch Constraint will obviously be “empty”. But I’m not sure what “empty” actually means in the context of Pyomo. If you wanted all indices represented, I would probably have set the index to Constraint.Feasible. I mainly point this out because I’m not entirely sure what the default constraint-data looks like in Pyomo. If you don’t want all indices represented, you could obviously just tweak con_set before the constraint is constructed.

Reply:
At the very beginning of the model definition for these simple OPF and OPF approximation models, we call "md = model_data.clone_in_service()" which effectively only includes the bus pairs here for active lines. Effectively, we are modifying the model prior to constructing all the constraints. This is ok for this simple single-stage model

Reply:
This makes sense - especially after you cull the code referenced in the comment below.